[WIP] feat(Regex): add custom Regex class and start entertaining regex flags

adnrs96 opened the pull request:
This adds a new class which will handle regex and provide methods to deal with regex and flags.

adnrs96 commented:
@judepereira can you check out the direction I am taking here and let me know what you think

judepereira commented:
Looks good. Let's add tests and merge it :)

wilzbach commented on storyruntime/utils/types/RegExp.py:
```diff
@@ -0,0 +1,57 @@
+from storyscript.compiler.semantics.types.Types import RegExpType
+
+
+class RegExp(RegExpType):
```
No. It should inherit from a to-be-created `RuntimeVar`. You want to use encapsulation here.

wilzbach replied:
For reference: https://github.com/storyscript/runtime/issues/327

adnrs96 commented:
@judepereira @wilzbach I updated this PR which is definitely meant to serve as a immediate solution.
In long term we should discuss the regex and how it should behave. And with that my earlier approach of defining a custom type for regex would definitely make sense but for now this should work.

Also storyscript/storyscript#1230 is the issue for dropping some of the flag support in compiler itself.

wilzbach commented on storyruntime/utils/RegExpUtils.py:
```diff
@@ -0,0 +1,18 @@
+
+    @staticmethod
+    def process_flags(flags):
+        re_flags = 0
```
thinking about we should probably be defensive here and do:

```py
all(x in "ims" for x in flags)
```

adnrs96 commented:
@judepereira Addressed the comment and rebased.

Though @wilzbach I later realized that this is one of those things which would endup error in compilation phase since we won't parse invalid flags in the first place
